storage/akamaistorage.py

```diff
@@ -69,6 +80,26 @@ def get_direct_download_url(
 
         # add akamai signed token
         try:
+            if self.amz_encryption_token is not None and self.amz_encryption_iv is not None:
+                query_params = urllib.parse.parse_qs(akamai_url_parsed.query)
+                amz_signature_params = query_params.get("X-Amz-Signature", [])
+                amz_signature = amz_signature_params[0] if len(amz_signature_params) == 1 else None
```

Comment:
can you check `encryption.py` to see if you can re-use those functions?

Reply:
hm kind of. It uses AEM-CCM which we need to see if it's supported by akamai. The methods are also specific for the db but i can modify them to make something work if need be

Reply:
akamai doesn't have an option for CCM encryption mode